### PR TITLE
Web: Document SPF navigation events and life cycle.

### LINKED ISSUE #52: Document SPF events and navigation lifecycle
SPF triggers important events that would be nice to have handy on `README.md`

### PR DESCRIPTION
**Pull request by nicksay**
Closes #52.

### CODE REVIEW

File: web/documentation/events.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+This event is fired before a request is sent for all types of navigation:
+clicks, back/forward, and API calls.  (Note that this event is fired even if a
+response is fetched from cache and no actual network request is made.)
+
+**`spfprocess`**  
+Fired when a response has been received, either from the network or from
+cache, before it is processed.  Use to update UI feedback (i.e. advance a
+progress bar) and dispose event listeners before content is updated.
+
+**`spfdone`**
```
Comment (DavidCPhillips) [inline web/documentation/events.md]: What about the mutlipart events? Are you planning on limiting those to the multipart page?

Comment (nicksay) [inline web/documentation/events.md]: Yes, I was planning on putting the multipart events on the multipart page and the script/style events on the resources page, and then I was going to provide links from the bottom of this page to those.  WDYT?

Comment (DavidCPhillips) [inline web/documentation/events.md]: Alright, that's reasonable.

**Comment by DavidCPhillips** [comment]
LGTM

**Pull request merged**